conda_forge_tick/make_graph.py

```diff
@@ -89,16 +90,20 @@ def make_graph(names, gx=None):
     total_names = new_names + old_names
     logger.info("start loop")
 
-    for i, name in enumerate(total_names):
-        try:
-            sub_graph = get_attrs(name, i)
-        except Exception as e:
-            logger.warn("Error adding {} to the graph: {}".format(name, e))
-        else:
-            if name in new_names:
-                gx.add_node(name, **sub_graph)
+    with ThreadPoolExecutor(max_workers=20) as pool:
+        futures = {pool.submit(get_attrs(name, i)): name for i, name in enumerate(total_names)}
+
+        for f in as_completed(futures):
+            name = futures[f]
```

Comment:
What is this line doing?

Reply:
This is getting the name of the node who's future has been completed. This is needed for the logger/exception so that we know which node failed.